docs/1.2-DRAFT/data-entities.md

```diff
@@ -214,6 +220,11 @@ Additionally, `Dataset` entities SHOULD have:
 
 Any of the properties of schema.org [Dataset] MAY additionally be used (adding contextual entities as needed). [Directories on the web](#directories-on-the-web-dataset-distributions) SHOULD also provide `distribution`.
 
+## Local Data Entities
+
+Where an RO-Crate is used to describe _files_  and _directories_ contained within the _RO-Crate root_ directory these locally referenced _Data Entities_ as described in the _RO-Crate Metadata Document_  SHOULD be present in the _RO-Crate root_.
```

Comment:
Should we say MUST here? I think the python library interprets this as a MUST -- we have not made our intent clear in the spec...

Reply:
No, the Python library allows data entities to be missing from the directory. This was requested by you in https://github.com/ResearchObject/ro-crate-py/issues/73. See also https://github.com/ResearchObject/ro-crate-py/issues/135, where changes are made so that such crates are "not the norm".

Reply:
Ok, yes, software libraries need to be able to read things that are not fully valid, so they can report warnings and errors -- the point I was making that the python implementers assumed that if you reference a file it SHOULD/MUST be there (which is perfectly reasonable) - the original point of the spec was to describe files -- it doesn't make sense to be shipping crates that contain references to things that aren't there.

Reply:
For  RO-Crate version 2 we may want to formally define what constitutes and error or a warning.